src/main/java/jenkins/plugins/office365connector/CardBuilder.java

```diff
@@ -194,41 +189,33 @@ public Card createBuildMessageCard(StepParameters stepParameters) {
     }
 
     private void addScmDetails() {
-        Set<User> users;
-        List<ChangeLogSet<ChangeLogSet.Entry>> sets;
-
-        try {
-            users = (Set<User>) run.getClass().getMethod("getCulprits").invoke(run);
-            sets = (List<ChangeLogSet<ChangeLogSet.Entry>>) run.getClass().getMethod("getChangeSets").invoke(run);
-        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
-            users = Collections.emptySet();
-            sets = Collections.emptyList();
+        if (!(run instanceof AbstractBuild)) {
```

Comment:
`AbstractBuild` does not cover workflow build jobs.
Please include a test for `WorkflowRun`: https://github.com/jenkinsci/workflow-job-plugin/blob/master/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

Reply:
Will check - thanks!

Reply:
I wonder if this is the only case if not, getting rid off reflection is hardly possible :(

Reply:
@damianszczepanik you bumped to Java 8 without bumping Jenkins core version.
https://github.com/jenkinsci/office-365-connector-plugin/blob/master/pom.xml#L28

2.60 is when Java 8 is available.

So if you bump Jenkins core you should have access to RunWithSCM and than you could drop reflection :smile:

Reply:
than it would be if `run instanceof RunWithSCM` and AbstractBuild and WorkflowRun both support that.

Reply:
I'd still suggest adding a test case for this piece for both Abstract and Workflow runs

Reply:
@casz  please review

Reply:
Should look something like this:
```java
        if (!(run instanceof RunWithSCM)) {
            return;
        }
        RunWithSCM runWithSCM = (RunWithSCM) run;
        factsBuilder.addCulprits(runWithSCM.getCulprits());

        List<ChangeLogSet<ChangeLogSet.Entry>> sets = runWithSCM.getChangeSets();
```

See https://github.com/jenkinsci/jenkins/blob/399c8ba40ab9ccab8e70dcacfdd396b0c2e654d7/core/src/main/java/hudson/model/View.java#L727-L754